pinot-plugins/pinot-stream-ingestion/pinot-kafka-3.0/src/main/java/org/apache/pinot/plugin/stream/kafka30/KafkaPartitionLevelConnectionHandler.java

```diff
@@ -78,16 +80,38 @@ private Properties buildProperties(StreamConfig streamConfig) {
       consumerProp.put(ConsumerConfig.ISOLATION_LEVEL_CONFIG, _config.getKafkaIsolationLevel());
     }
     consumerProp.put(ConsumerConfig.CLIENT_ID_CONFIG, _clientId);
+    if (consumerProp.containsKey(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG)
+        && consumerProp.getProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG).equals("smallest")) {
```

Comment:
Is this needed? If so, should we apply the same change to Kafka 2.0 class?

Reply:
It's not relevant to this change here. I was doing some other testing. But we may want to take it in too as part of a different PR?